eng/scripts/Array-Functions.ps1

```diff
@@ -0,0 +1,18 @@
+function ShardArray($array, $shard, $totalShards) {
+    if ($totalShards -lt 2) { 
+      return $array
+    }
+
+    if ($shard -ge $totalShards) {
+      throw "Shard index ($shard) must be less than total shards ($totalShards)"
+    }
+
+    if ($totalShards -gt $array.Length) {
+      throw "Cannot shard array into more pieces than there are elements"
```

Comment:
Would we hit this case if there were only one spec discovered but we have it sharded in 3? I assume this wouldn't be the case for tsv-all but could be in other cases if we use this logic.

Reply:
Yes, currently the caller needs to check this first:

https://github.com/Azure/azure-rest-api-specs/pull/31033/files/1fbda3c72ab85858bea20a6a2f9c3654b45b49ed#diff-608a3bb7e5c6d9a396eb1c6636c575f534ed9d9649926eaf2249880c995a4db0R25